core/src/main/scala/cats/syntax/either.scala

```diff
@@ -194,6 +194,12 @@ final class EitherOps[A, B](private val eab: Either[A, B]) extends AnyVal {
       case r @ Right(_) => EitherUtil.leftCast(r)
     }
 
+  def leftMapOrKeep[AA >: A](pf: PartialFunction[A, AA]): Either[AA, B] =
+    eab match {
+      case Left(a)      => Left(pf.applyOrElse(a, identity[AA]))
+      case r @ Right(_) => r
```

Comment:
I wonder if on a byte-code level this would be more performant:
```
case r : Right[B] => r
```
because there's no extractor to be called nor intermediate `Option` value to be created in opposite to the former case.

Reply:
Interestingly, I never thought that `x @ Foo(_)` might have any significant distinction from `x: Foo`.